src/index.ts

```diff
@@ -1,333 +1,251 @@
-    deviceOwnershipCode,
-  );
-
-export const onboardDevice = ({
-  deviceId,
-  certsResponse,
-  apiHost,
-  apiKey,
-  verbose,
-}: Partial<SimulatorConfig>) => {
-  let certificate = JSON.parse(certsResponse as string).clientCert;
-  return getConn(
-    apiHost as string,
-    apiKey as string,
-    !!verbose,
-  ).post(`v1/devices/${deviceId}`, { certificate });
-};
+  preventAssociation: boolean;
+  jobExecutionFailureScenario?: JobExecutionFailureScenario;
+  verbose: boolean;
+}
+export interface DeviceConfig extends ConfigCommon {
+  deviceCredentials: DeviceCredentials;
+  mqttEndpoint: string;
+  mqttTopicPrefix: string;
+  mqttMessagesTopicPrefix: string;
+  sensors: Map<string, ISensor>;
+}
+export interface SimulatorConfig extends ConfigCommon {
+  apiKey: string;
+  apiHost: string;
+  sensors?: string[];
+}
+
+export const run = async (simConfig: SimulatorConfig): Promise<void> => {
+  const {
+    deviceType,
+    certificateType,
+    preventAssociation,
+    apiKey,
+    apiHost,
+    verbose,
+  } = simConfig;
 
-export const getDefaults = async ({
-  deviceId,
-  endpoint,
-  mqttMessagesPrefix,
-  certsResponse,
-  apiHost,
-  apiKey,
-  deviceOwnershipCode,
-  verbose,
-  onboard,
-}: Partial<SimulatorConfig>): Promise<DeviceDefaults> => {
-  const conn = getConn(apiHost!, apiKey!, !!verbose);
   const log = new Log(!!verbose);
-
-  const defaults: DeviceDefaults = {
-    endpoint: endpoint || '',
-    mqttMessagesPrefix: mqttMessagesPrefix || '',
-    certsResponse: certsResponse || '',
-    tenantId: '',
-  };
-
-  const cacheFile = cache.getFilePath(deviceId);
-
-  const cachedDefaults: DeviceDefaults = cache.exists(cacheFile)
-    ? await cache.get(cacheFile)
-    : {};
-
-  if (!(endpoint && mqttMessagesPrefix)) {
-    log.debug(`Grabbing mqttEndpoint and messagesPrefix...`);
-    let defaultEndpoint = cachedDefaults.endpoint || '',
-      defaultMqttMessagesPrefix = cachedDefaults.mqttMessagesPrefix || '';
-
-    if (!(defaultEndpoint && defaultMqttMessagesPrefix)) {
-      log.debug('Fetching endpoints from device API.\n');
-      const { data } = await conn.get(`/v1/account`);
-      defaultMqttMessagesPrefix = data.mqttTopicPrefix + 'm/';
-      defaultEndpoint = data.mqttEndpoint;
-    }
-
-    if (!endpoint) {
-      defaults.endpoint = defaultEndpoint;
+  const restApiClient = new RestApiClient(apiHost, apiKey, !!verbose);
+  const hostSplit = simConfig.apiHost.split('.');
+  const stage = hostSplit.length === 3 ? 'prod' : hostSplit[1];
+  const teamData = await restApiClient.fetchTeamInfo();
+  const teamId = teamData.team.tenantId;
+  const teamName = teamData.team.name;
+  const mqttTopicPrefix = `${stage}/${teamId}`;
+  const mqttEndpoint = `mqtt${stage === 'prod' ? '' : `.${stage}`}.nrfcloud.com`;
+  let deviceId = simConfig.deviceId;
+  if (deviceId) {
+    simConfig.deviceCredentials = getLocallyStoredDeviceCredentials(deviceId, log);
+    if (!simConfig.deviceCredentials) {
```

Comment:
So if someone onboards, deletes, then re-onboards a device, will it use the old credentials? Even if they onboard it with a different tenant (API key)?